OpenRA.Game/Network/SyncReport.cs

```diff
@@ -298,12 +299,12 @@ public Values(int size)
 				}
 			}
 
-			public object this[int index]
+			public object? this[int index]
 			{
 				get
 				{
 					if (item2OrSentinel == Sentinel)
-						return ((object[])item1OrArray)[index];
+						return ((object?[])item1OrArray!)[index];
```

Comment:
I hope in general to employ the null-forgiving operator (`!`) as little as possible. But here's an example where I think it makes sense. The code has a local invarient where if item 2 is the sentinel, then the first item is an array. The compiler doesn't know this but since we've checked locally supressing the potiental null is reasonable.